storage-proofs/post/src/fallback/vanilla.rs

```diff
@@ -380,16 +383,28 @@ impl<'a, Tree: 'a + MerkleTreeTrait> ProofScheme<'a> for FallbackPoSt<'a, Tree>
                     Tree::Arity::to_usize(),
                 );
 
+                let num_challenges = if pub_params.is_winning {
+                    pub_params.sector_count
```

Comment:
I think this depends on `pub_params.challenge_count` being equal to 1. Since that's not structurally guaranteed (but will be the case in extant WindowPoSt), I think it would be good to also assert it here. I'd rather be safe than sorry, at least until we can clean this up more broadly.